File: procon-front/src/app/utils/mascaras-utils.ts
export function getMascaraFone(value: string): string {
  if (value) {
    if (value.charAt(0)) {
      if (value.charAt(0) === '1') return '000 00';
      if (value.charAt(0) === '4')
        if (value.charAt(1) && value.charAt(1) === '0') return '0000-0000';
    }
    if (value.charAt(1)) {
      if (value.charAt(1) === '8' && value.charAt(0) === '0')
        return '0000 000 0000';
    }
    if (value.charAt(2)) {
      if (value.charAt(2) === '9') return '(00) 00000-0000';
    }
    return '(00) 0000-0000';
  }
  return value;
}

File: procon-api/src/main/java/br/com/procon/utils/MascarasUtils.java
package br.com.procon.utils;

import java.text.ParseException;

import javax.swing.text.MaskFormatter;

/**
 * 
 * @author Marlon F. Garcia
 *
 */
public class MascarasUtils {

	/**
	 * Aplica máscara a uma string.
	 * 
	 * @param pattern
	 * @param value
	 * @return
	 */
	public static String format(String pattern, String value) {
		try {
			MaskFormatter mask;
			String tratada = value.replaceAll("[-()]*", "");
			mask = new MaskFormatter(pattern);
			mask.setValueContainsLiteralCharacters(false);
			return mask.valueToString(tratada);
		} catch (ParseException e) {
			return value;
		}
	}

	/**
	 * Aplica máscaras aos telefones.
	 * 
	 * @param value
	 * @return
	 */
	public static String foneFormat(String value) {
		try {
			if (value.charAt(2) == '9')
				return format("(##) #####-####", value);
			else
				return format("(##) ####-####", value);
		} catch (Exception e) {
			e.printStackTrace();
			return value;
		}
	}

}
